extensions/amp-ad/0.1/legacy-ad-intersection-observer-host.js

```diff
@@ -247,13 +263,19 @@ export class LegacyAdIntersectionObserverHost {
    * Sends 'intersection' message to ad/iframe with intersection change records
    * if this has been activated and we measured the layout box of the iframe
    * at least once.
+   * @param {!IntersectionObserverEntry} entry - handed over by the IntersectionObserver.
    * @private
    */
-  sendElementIntersection_() {
-    if (!this.intersectionObserver_) {
-      return;
-    }
-    const change = this.baseElement_.element.getIntersectionChangeEntry();
+  sendElementIntersection_(entry) {
+    // TODO(#31707): IntersectionObserverEntries cannot be serialized.
+    const change = {
+      time: entry.time,
+      rootBounds: entry.rootBounds,
```

Comment:
TBH, I'm even surprised that DOMRect is serializable. Would you mind checking this with Safari/FF?

Reply:
Do you have any recommendations for how to serialize? Should I do it all manually?

Reply:
I usually try not to risk it and serialize manually unless the API explicitly says that the structure is serializable. Annoying, but avoids nasty surprises.

Reply:
It [is](https://www.w3.org/TR/geometry-1/#structured-serialization) supposed to be serializable, but I'll manually write it out jic it doesn't have great browser support.

Reply:
If it's explicitly promised to be serializable, then it's all good. We could always check on IE, but I'd expect they followed it that far.